client/gefyra/cluster/resources.py

```diff
@@ -158,18 +158,51 @@ def create_operator_deployment(
 
 
 def get_pods_and_containers_for_workload(
+    config: ClientConfiguration, name: str, namespace: str, workload_type: str
+) -> Dict[str, List[str]]:
+    workload = None
+    result = {}
+    if workload_type == "deployment":
+        workload = config.K8S_APP_API.read_namespaced_deployment(
+            name=name, namespace=namespace
+        )
+    elif workload_type == "statefulset":
+        workload = config.K8S_APP_API.read_namespaced_stateful_set(
+            name=name, namespace=namespace
+        )
+
+    if not workload:
+        logger.error(f"Could not find {workload_type} - {name}.")
+        exit(1)
```

Comment:
Please raise a `RuntimeError` instead of exiting
Note: maybe this branch is not required if 404